### PR TITLE
Fix shape of bias tensor in pytorch if zero, add additional batchnorm tests

### PR DESCRIPTION
**Pull request by JanFSchulte**
Fixes the bug reported in https://github.com/fastmachinelearning/hls4ml/pull/1045, where the incorrect tensor shape was used to generate zeros for bias tensors in case of pytorch models. This version of the fix includes a protection for layers that don't have the `data_format` attribute and a clean version of the tests for batchnorms introduced in that PR.  

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [x] Bug fix (non-breaking change that fixes an issue)

## Tests

Added new pytests for Batchnorm layers in pytorch. Verified that rest of the pytorch tests still pass. 

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/main/CONTRIBUTING.md).
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have installed and run `pre-commit` on the files I edited or added.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by vloncar** [comment]
Can we not mix `'` and `"`. Most of the code uses `'` :wink:
